sdk/core/azure-core-http-jdk-httpclient/src/main/java/com/azure/core/http/jdk/httpclient/JdkAsyncHttpClientBuilder.java

```diff
@@ -144,7 +182,56 @@ public HttpClient build() {
                         buildProxyOptions.getPassword()));
             }
         }
-        return new JdkAsyncHttpClient(httpClientBuilder.build());
+        return new JdkAsyncHttpClient(httpClientBuilder.build(), Collections.unmodifiableSet(getRestrictedHeaders()));
+    }
+
+    Set<String> getRestrictedHeaders() {
+        // Compute the effective restricted headers by removing the allowed headers from default restricted headers
+        Set<String> allowRestrictedHeaders = getAllowRestrictedHeaders();
+        Set<String> restrictedHeaders = new TreeSet<>(String.CASE_INSENSITIVE_ORDER);
```

Comment:
What's the purpose of having Case_Insensitive_Order?

Reply:
Case insensitive order will allow for `set.contains()` to ignore case and evaluate if the header is restricted or not. See unit tests.